dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/action/validation/NotificationProgramRuleActionValidator.java

```diff
@@ -64,26 +64,22 @@ public ProgramRuleActionValidationResult validate(
 
     if (pnt == null) {
       // fetch it from database
-      pnt =
-          validationContext
-              .getProgramRuleActionValidationService()
-              .getNotificationTemplateService()
-              .getByUid(programRuleAction.getTemplateUid());
+      pnt = programRuleAction.getNotificationTemplate();
     }
 
     if (pnt == null) {
       log.debug(
           String.format(
               "ProgramNotificationTemplate id: %s for program rule: %s does not exist",
-              programRuleAction.getTemplateUid(), rule.getName()));
+              programRuleAction.getNotificationTemplate().getUid(), rule.getName()));
```

Comment:
This is a NullPointerException.
And no failing test pointing this issue means that we are not testing the case where the notification template is not present

Reply:
good catch @enricocolasante. After this change, i think we won't be needing this part of the code. 
I will add test as well to verify this change.